File: src/renderer.zig
const gl = @import("gl");

const root = @import("root.zig");
const VertexArray = root.VertexArray;
const IndexBuffer = root.IndexBuffer;
const Shader = root.Shader;

pub const Renderer = struct {
    pub fn init() Renderer {
        return Renderer{};
    }

    pub fn clear(_: Renderer) void {
        // gl.ClearColor(0.0, 0.0, 0.0, 1.0);
        gl.Clear(gl.COLOR_BUFFER_BIT);
    }

    pub fn draw(_: Renderer, vao: VertexArray, ibo: IndexBuffer, shader: Shader) void {
        shader.bind();
        vao.bind();
        ibo.bind();
        gl.DrawElements(gl.TRIANGLES, @intCast(ibo.count), gl.UNSIGNED_INT, 0);
    }
};

File: src/renderer.cpp
#include "renderer.h"

#include "glad/glad.h"

void renderer_clear() {
    glClear(GL_COLOR_BUFFER_BIT);
    glClearColor(0, 0, 0, 1);
}

void renderer_draw(struct Shader *shader, struct VertexArray *vao, struct IndexBuffer *ibo) {
    shader_bind(shader);
    vao_bind(vao);
    ibo_bind(ibo);
    glDrawElements(GL_TRIANGLES, ibo->count, GL_UNSIGNED_INT, 0);
}
